src/test/java/it/com/atlassian/bitbucket/jenkins/internal/fixture/BitbucketJenkinsRule.java

```diff
@@ -129,6 +134,12 @@ public void before() throws Throwable {
                 "", BITBUCKET_ADMIN_USERNAME, ADMIN_PERSONAL_TOKEN.get().getSecret());
         addCredentials(readCredentials);
 
+        sshCredentialId = UUID.randomUUID().toString();
```

Comment:
Interested in feedback in this class in particular. We are relying on server configuration to get credentials for the SCM elsewhere in the code, so to prevent having each class that wants to do an SSH test manually parsing through the credential store, I've added a field for it. We may want to reconsider this.

Reply:
How expensive is it to create an SSH key? The reason we're doing it for the accesstoken is that it takes 1+ seconds to create one, which makes the tests run longer for no good reason. I'm not convinced (but I admit I don't know) that adding SSH credentials to Jenkins would be so slow?

Reply:
I believe creating SSH keys is much faster than that. Are you referring to having the SSH credentials in the rule class at all, or creating and deconstructing them in a thread?